source/Helpers/POENinjaAPI/PoeNinjaClient.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Newtonsoft.Json;
+using Sidekick.Helpers.POENinjaAPI.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Sidekick.Helpers.POENinjaAPI
+{
+    public class PoeNinjaClient
+    {
+
+        public readonly static Uri POE_NINJA_ITEMOVERVIEW_BASE_URL = new Uri("https://poe.ninja/api/data/itemoverview");
+        public readonly static Uri POE_NINJA_CURRENCYOVERVIEW_BASE_URL = new Uri("https://poe.ninja/api/data/currencyoverview");
+        private readonly HttpClient _client;
+
+        public PoeNinjaClient()
+        {
+            _client = new HttpClient();
```

Comment:
DISCLAIMER: I haven't downloaded the source code yet, but I would personally recommend not instantiating HttpClient here. 

It is highly recommended to only instantiate HttpClient once throughout the application, and the way it is currently implemented multiple instances could be created which could lead to a common problem known as "socket exhaustion". See https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=netframework-4.8

What i would do is to create an static HttpClient provider, which would create a HttClient instance on its first call and hold that instance untill the application is terminated.

I will download the source code and take a look to see what i can do later.

Reply:
I totally agree! If someone wants to tackle this feel free to do so. I won't get to it for probably a week otherwise I would do it.

We should also probably share a HttpClient for both the poe.ninja and the official Poe trade API.

Reply:
Yes, since I think .NET 4.5, the HttpClient can be used everywhere and be thread-safe with concurrent requests. We would just need to change the way the base uri is set and take out the client from the TradeClient.